libraries/aws_backend.rb

```diff
@@ -278,6 +292,28 @@ def map_tags(tag_list)
     end
     tags
   end
+
+  private
+
+  # This method should be used when AWS API returns multiple resources for the provided criteria.
+  def resource_fail(message = nil)
+    message ||= "#{@__resource_name__}: #{@display_name}. Multiple AWS resources were returned for the provided criteria. "\
+    'If you wish to test multiple entities, please use the plural resource. '\
+    'Otherwise, please provide more specific criteria to lookup the resource.'
```

Comment:
On the fence, whether it would be better to make `message` a required method parameter, rather than defaulting to this message.

Reply:
The idea behind this is that returning a standard message.
It can still be overridden if there is something more specific to the issue.